pkg/registry/generic/etcd/etcd.go

```diff
@@ -441,6 +441,9 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	// delete immediately, or no graceful deletion supported
 	out := e.NewFunc()
 	if err := e.Storage.Delete(ctx, key, out); err != nil {
+		if storage.IsNotFound(err) {
```

Comment:
I think this should only be tolerated if we went through the `graceful` path above